### ISSUE TITLE
Failure to deploy VM with 2 or more attached SATA disks.

### ISSUE THREAD

**Issue by lkolator** [OpenNebula/one#5705]
**Description**
When try deploy VM with 2 or more SATA disks attached I got error:
`
DEPLOY: error: Failed to create domain from /var/lib/one/datastores/101/333/deployment.0 error: unsupported configuration: target must be 0 for ide controller Could not create domain from /var/lib/one/datastores/101/333/deployment.0 ExitCode: 255
`

**To Reproduce**

1. Create two images with BUS: SCSI/SATA
2. Set in VM template: OS & CPU -> Boot ->Bus for SD disks: SATA
3. Attach 2 SD disks in VM template storage section.
4. Instantiate VM template.

**Expected behavior**
Should be possible to attach more than one SATA disk.

**Details**
 - Affected Component: KVM Driver
 - Hypervisor: KVM
 - Version: 5.12.0.4, 6.2

**Additional context**
Deployment error is caused from that OpenNebula KVM Driver create deployment config with SCSI controller always
when are 2 or more SD disks(regardless than  `Bus for SD disks` is SCSI or SATA):
```
deployment.0:
...
                <controller type='scsi' index='0' model='virtio-scsi'>
                        <driver queues='1'/>
                </controller>
...
```
When try to add one SATA disk opennebula doesn't paste this controller config to deployment file and libvirt automatically add needed SATA controller to VM.

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by jirutka** [comment]
The same issue was reported 3 years before this one in #2971. @sergiojvg closed it as fixed, but it’s still not fixed.

**Comment by rsmontero** [comment]
Hi @jirutka 

Could you share the VM template or VM data, like onetemplate show -j <template> or onevm show -j <vmname>. So we can work on the specific configuration triggering the issue?

Cheers

**Comment by jirutka** [comment]
```json
{
	"VMTEMPLATE": {
	  "ID": "666",
	  "UID": "66",
	  "GID": "0",
	  "UNAME": "flynn",
	  "GNAME": "oneadmin",
	  "NAME": "nebulshit",
	  "PERMISSIONS": {
		"OWNER_U": "1",
		"OWNER_M": "1",
		"OWNER_A": "0",
		"GROUP_U": "0",
		"GROUP_M": "0",
		"GROUP_A": "0",
		"OTHER_U": "0",
		"OTHER_M": "0",
		"OTHER_A": "0"
	  },
	  "REGTIME": "1698353975",
	  "TEMPLATE": {
		"CONTEXT": {
		  "NETWORK": "YES",
		  "SSH_PUBLIC_KEY": "$USER[SSH_PUBLIC_KEY]"
		},
		"CPU": "2",
		"CPU_MODEL": {
		  "MODEL": "host-passthrough"
		},
		"DISK": [
		  {
			"CACHE": "none",
			"DEV_PREFIX": "sd",
			"IMAGE": "nebulshit-disk-1",
			"IMAGE_UNAME": "some-group",
			"IO": "native"
		  },
		  {
			"CACHE": "none",
			"DEV_PREFIX": "sd",
			"IMAGE": "nebulshit-disk-2",
			"IMAGE_UNAME": "some-group",
			"IO": "native"
		  },
		  {
			"CACHE": "none",
			"DEV_PREFIX": "sd",
			"IMAGE": "nebulshit-disk-3",
			"IMAGE_UNAME": "some-group",
			"IO": "native"
		  }
		],
		"FEATURES": {
		  "ACPI": "yes",
		  "GUEST_AGENT": "no",
		  "IOTHREADS": "1",
		  "VIRTIO_SCSI_QUEUES": "1"
		},
		"GRAPHICS": {
		  "LISTEN": "0.0.0.0",
		  "TYPE": "VNC"
		},
		"HOT_RESIZE": {
		  "CPU_HOT_ADD_ENABLED": "NO",
		  "MEMORY_HOT_ADD_ENABLED": "NO"
		},
		"HYPERVISOR": "kvm",
		"INPUT": [
		  {
			"BUS": "usb",
			"TYPE": "tablet"
		  }
		],
		"MEMORY": "1024",
		"MEMORY_RESIZE_MODE": "BALLOONING",
		"MEMORY_UNIT_COST": "MB",
		"OS": {
		  "ARCH": "x86_64",
		  "BOOT": "disk1",
		  "FIRMWARE": "/usr/share/OVMF/OVMF_CODE.fd",
		  "MACHINE": "pc-q35-5.2",
		  "SD_DISK_BUS": "sata"
		},
		"SCHED_REQUIREMENTS": "ID=\"0\" | ID=\"1\"",
		"TOPOLOGY": {
		  "CORES": "2",
		  "PIN_POLICY": "THREAD",
		  "SOCKETS": "1",
		  "THREADS": "2"
		},
		"VCPU": "4"
	  }
	}
  }
```

**Comment by rsmontero** [comment]
OK, Thanks, we'll look into it

**Comment by rsmontero** [comment]
Fixed as per 592a2a64f555e09.

The patch adds the needed SATA Controllers when using sata bus, it considers if machine type is q35 as it adds a ACHI controller in slot 0x1f function 2 (this one is used for context disk in this case).
